server.py

```diff
@@ -0,0 +1,343 @@
+        print "No email exist"
+        return True
+    else:
+        print "email exist"
+        return False
+
+
+@app.route('/email_availability/', methods=['GET'])
+def email_availability():
+    """
+    Email availability route
+    """
+    email = request.args.get('email')
+    print(email)
+    if email_existence(str(email)) is True:
+        return "true"
+    else:
+        return "false"
+
+
+def register_user(name, password, email):
+    """
+    This function is registering the user.
+    :param name: Name of user
+    :param password: Password of user
+    :param email: Email of user
+    """
+    name = str(name)
+    password = str(password)
+    email = str(email)
+    new_user = Users(name=name, password=password, email=email)
+    db.session.add(new_user)
+    db.session.commit()
+
+
+@app.route('/login', methods=['GET'])
+def login():
+    """
+    Login route.
+    """
+    return render_template("login.html")
+
+
+def get_authenticated(email, password):
+    """
+    This function is authenticating user for Login.
+    :param email: Email of user
+    :param password: Password
+    :return: Status of authentication.
+    """
+    result = Users.query.filter(and_(Users.email == str(email), Users.password == str(password))).all()
+    if result.__len__() < 1:
+        return False
+    else:
+        session['user'] = result[0].id
+        return True
+
+
+@app.route('/layout', methods=['GET'])
+def layout():
+    """
+    Layout route
+    :return:
+    """
+    if 'user' in session:
+        return render_template("layout.html")
+    else:
+        return redirect(url_for('login'))
+
+
+@app.route('/load_data', methods=['GET'])
+def load_data():
+    """
+    Load data route. This function is loading data on layout page after loading.
+    :return: Returns all images
+    """
+    image_list = get_image_data()
+    print(image_list)
+    return jsonify(image_list)
+
+
+@app.route('/login/', methods=['GET'])
+def authentication():
+    """
+    Login route for authentication
+    :return: Returns Login status
+    """
+    result = get_authenticated(request.args.get('email'), request.args.get('password'))
+    if result is True:
+        print("authenticated")
+        return "true"
+    else:
+        return "false"
+
+
+@app.route('/signup', methods=['GET'])
+def signup_page():
+    """
+    Rendering signup page.
+    :return:
+    """
+    return render_template("signup.html")
+
+
+@app.route('/signup/', methods=['POST'])
+def signup():
+    """
+    Signup route for registration.
+    """
+    form_data = request.form
+    register_user(form_data.get('user_name'), form_data.get('password'), form_data.get('email'))
+    return redirect(url_for('login'))
+
+
+def delete_image(image_id):
+    """
+    This function is deleting image. Accepts image id from delete route and deletes image.
+    :param image_id:
+    :return:
+    """
+    image_id = int(image_id)
+    Images.query.filter(Images.id == image_id).delete()
```

Comment:
I think we should check if image belongs to logged-in user, then we delete it. Also remove inline query and session commit code.